src/test/java/org/candlepin/subscriptions/product/OfferingSyncControllerTest.java

```diff
@@ -108,4 +146,157 @@ void testSyncOfferingNoProductIdsShouldPersist() {
     // Then it should still persist, since there are Offerings that we need that have no eng prods.
     verify(repo).save(sku);
   }
+
+  @Test
+  void testGetUpstreamOfferingForOcpOffering() {
```

Comment:
Rather than feeding the tests fake product data, the tests use real product data, stripped down to only the fields and attrs that are actually used in creating an Offering. There are benefits to this, but again if there are strong opinions against this technique then let me know.